Modules/Quotas/test/Quotas.Application.Tests/Tests/Tiers/GetTierById/HandlerTests.cs

```diff
@@ -39,8 +39,64 @@ public async void Gets_tier_by_id()
         result.Quotas.First().Period.Should().Be(period);
     }
 
-    private Handler CreateHandler(ITiersRepository tiers, IMetricsRepository metrics)
+    [Fact]
+    public async void Gets_tier_by_id_multiple_quotas()
+    {
+        // Arrange
+        var tierId = new TierId("SomeTierId");
+        var tierName = "some-tier-name";
+        var tier = new Tier(tierId, tierName);
+
+        var metric1 = new Metric(MetricKey.NumberOfSentMessages, "Number Of Sent Messages");
+        var metric2 = new Metric(MetricKey.FileStorageCapacity, "File Storage Capacity");
+        var metric3 = new Metric(MetricKey.NumberOfRelationships, "Number Of Relationships");
+        var metrics = new List<Metric> { metric1, metric2, metric3 };
+
+        tier.CreateQuota(metric1.Key, 5, QuotaPeriod.Month);
+        tier.CreateQuota(metric2.Key, 10, QuotaPeriod.Day);
+        tier.CreateQuota(metric3.Key, 12, QuotaPeriod.Week);
+        tier.CreateQuota(metric1.Key, 3, QuotaPeriod.Year);
+
+        var stubTiersRepository = new FindTiersStubRepository(tier);
+        var stubMetricsRepository = new FindAllWithKeysMetricsStubRepository(metrics);
+
+        var handler = CreateHandler(stubTiersRepository, stubMetricsRepository);
+
+        // Act
+        var result = await handler.Handle(new GetTierByIdQuery(tierId), CancellationToken.None);
+
+        // Assert
+        result.Id.Should().Be(tierId);
+        result.Name.Should().Be(tierName);
+        result.Quotas.Should().HaveCount(4);
+
+        foreach (var quota in result.Quotas)
```

Comment:
You can assert on the index of the quotas instead of having a loop:

```c#
tier.Quotas.ElementAt(0).Metric.Key.Should().Be(metricWithTwoQuotas.Key)
...
```